core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -656,7 +656,8 @@ public void doDelete(final List<String> topicsToDelete,
 
 
     private boolean isInternalTopic(final String topicName) {
-        return topicName.startsWith(options.valueOf(applicationIdOption) + "-")
+        return !isInputTopic(topicName) && !isIntermediateTopic(topicName)
```

Comment:
One nit: can we add a comment why we check for `isInputTopic(topicName)` -- it's a little unclear otherwise (maybe also add the link to the Jira so people can re-read the discussion)

Reply:
done